arelle/ModelXbrl.py

```diff
@@ -266,73 +191,82 @@ class ModelXbrl:
+        self.nameConcepts: defaultdict[str, list[ModelConcept]] = defaultdict(list)  # contains ModelConcepts by name
+        self.qnameAttributes: dict[QName, Any] = {}
+        self.qnameAttributeGroups: dict[QName, Any] = {}
+        self.qnameGroupDefinitions: dict[QName, Any] = {}
+        self.qnameTypes: dict[QName, ModelType] = {}  # contains ModelTypes by qname key of type
+        self.baseSets: defaultdict[Any, Any] = defaultdict(list)  # contains ModelLinks for keys arcrole, arcrole#linkrole
+        self.relationshipSets: dict[tuple[str] | tuple[str, tuple[str] | str | None, QName | None, QName | None, bool], ModelRelationshipSetClass] = {}  # contains ModelRelationshipSets by bas set keys
+        self.qnameDimensionDefaults: dict[str, str] = {}  # contains qname of dimension (index) and default member(value)
+        self.facts: list[ModelInlineFact] = []
+        self.factsInInstance: set[ModelInlineFact] = set()
+        self.undefinedFacts: list[ModelFact] = []  # elements presumed to be facts but not defined
+        self.contexts: dict[str, modeldoc.xmlRootElement] = {}
+        self.units: dict[str, ModelUnit] = {}
+        self.modelObjects: list[ModelObject] = []
+        self.qnameParameters: dict[QName, Any] = {}
+        self.modelVariableSets: set[Any] = set()
+        self.modelConsistencyAssertions: set[Any] = set()
+        self.modelCustomFunctionSignatures: dict[QName, Any] = {}
+        self.modelCustomFunctionImplementations: set[modeldoc] = set()
+        self.modelRenderingTables: Set[Any] = set()
         if not keepViews:
-            self.views = []
-        self.langs = {self.modelManager.defaultLang}
+            self.views: list[Any] = []
+        self.langs: set[str] = {self.modelManager.defaultLang}
         from arelle.XbrlConst import standardLabel
-        self.labelroles = {standardLabel}
-        self.hasXDT = False
-        self.hasTableRendering = False
-        self.hasTableIndexing = False
-        self.hasFormulae = False
+        self.labelroles: set[str] = {standardLabel}
+        self.hasXDT: bool = False
+        self.hasTableRendering: bool = False
+        self.hasTableIndexing: bool = False
+        self.hasFormulae: bool = False
         self.formulaOutputInstance = None
-        self.logger = logging.getLogger("arelle")
-        self.logRefObjectProperties = getattr(self.logger, "logRefObjectProperties", False)
-        self.logRefHasPluginAttrs = any(True for m in pluginClassMethods("Logging.Ref.Attributes"))
-        self.logRefHasPluginProperties = any(True for m in pluginClassMethods("Logging.Ref.Properties"))
-        self.logRefFileRelUris = defaultdict(dict)
-        self.profileStats = {}
-        self.schemaDocsToValidate = set()
-        self.modelXbrl = self # for consistency in addressing modelXbrl
-        self.arelleUnitTests = {} # unit test entries (usually from processing instructions
+        self.logger: Any = logging.getLogger("arelle")
```

Comment:
This is an example of where type information is lost. `logging.getLogger(...)` already has a return value, but it's discarded by the `Any` annotation.

```suggestion
        self.logger = logging.getLogger("arelle")
```